odoo/addons/base/models/res_lang.py

```diff
@@ -111,6 +116,48 @@ def _check_grouping(self):
             except Exception:
                 raise ValidationError(warning)
 
+    @api.depends('date_format')
+    def _compute_date_format_preview(self):
+        """ Sometimes language may not be activated and the format DateTime would not work as expected.
+        hence, we are directly using babel to parse the required language to format the date.
+        """
+        self.date_format_preview = False
+        for lang in self.filtered(lambda lang: lang.date_format and lang.code):
+            locale = tools.babel_locale_parse(lang.code)
+            try:
+                date_format = tools.posix_to_ldml(lang.date_format, locale=locale)
+                lang.date_format_preview = '(e.g. "' + babel.dates.format_datetime(fields.Datetime.now(), date_format, locale=locale) + '")'
+            except KeyError:
```

Comment:
instead of catching the exception, can't we check that `lang.code` is in the list ?
(and then remove the try / except)

same for other methods